03-hello-caqti-lwt/lib/repo/init.mli

```diff
@@ -0,0 +1,5 @@
+val connect :
+  unit ->
+  ((module Caqti_lwt.CONNECTION), [> Caqti_error.load_or_connect ]) result Lwt.t
```

Comment:
In accordance my comment on `module_of_conn`, note that there is always a type alias `type connection = (module CONNECTION)` wherever the `CONNECTION` signature is defined.  I wonder if it would be preferable to use this, since it might encourage thinking of connections as values, until used.  I suspect the use of first-class module in Caqti may be a weakness from a pedagogical point of view.  It could in fact be eliminated by providing a complete set of wrapper functions for the functions defined in the signature if this is correct.  For now, I suggest promoting an unpack-where-used style.

Reply:
Got it. You did tell me on the OCaml forum a few months ago but I missed that detail at the time.

I'll remove this confusing unpacking/repacking. 

 first-class modules is certainly a concept I haven't fully grasped yet